### DISCUSSION TITLE
Astro.props entry in the API Reference page?

### DISCUSSION THREAD

**Discussion by ekafyi** [withastro/docs#630] [Ideas]
Would it be a good idea to add an entry for `Astro.props` in https://docs.astro.build/en/reference/api-reference ?

While it has been briefly addressed in https://docs.astro.build/en/reference/api-reference/#data-passing-with-props, people may not know that eg. `Astro.props` contain the HTML element `tag` property, which can be used to render an element polymorphically, as discussed in https://discord.com/channels/830184174198718474/830184175176122389/977974602431086632

**Comment by delucis** [comment]
Great idea! We should definitely include `Astro.props` on that page.

About dynamic tag rendering: there's nothing “special” about a `tag` prop so I'm not sure this would go there. You could name your prop `is` or `as` for example and it would also work. All that needs to happen is for you to use a variable that is a string, so that could come from a prop or something else:

```astro
---
const El = 'div'
---
<El>Hello!</El>
```

**Comment by sarah11918** [comment]
This does seem like a huge ommision, since it's something EVERYBODY uses! Great call, @ekafyi !

I think this rises to the level of "Issue", if you'd like to make one so it gets on the task list.

**Reply by delucis** [reply]
Created #632 from @ekafyi’s initial post 👍

**Comment by ArmandPhilippot** [comment]
We're closing stale and outdated discussion to keep only the discussions still relevant to the current project version and/or the current docs state.

This seems to be resolved now, and this can be closed!

If you think this is a mistake and that this discussion should remain open, please feel free to ping me, I'll take a second look at it.

Thank your for your contribution!
